### PR TITLE
Buffs the Poisoned Knife

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Buffs the poisoned Knife by decreasing its TC cost and increasing it's reagent transfer amount to 5 or 10.

## Why It's Good For The Game
The poisoned combat knife barely sees any use due to it being an extremely subpar tool for injecting chemicals. Partially because it's heavily overshadowed by the sleepy pen, but that's another issue.
image
This PR aims to make the knife a more appealing option for the traitors who wish to engage in chemical warfare.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/08fe1c28-c967-4d2a-b322-2b2ceb7e0ab0

</details>

## Changelog
:cl:
balance: Decreased the price of the Poisoned Knife from 8 TC to 6
balance: Increased the maximum amount of reagents the poisoned knife can transfer per hit to 10
add: Added baloon alerts to changing the poison knife's transfer amounts, visible only to the user
/:cl:

### CODE REVIEW

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -446,9 +446,9 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 
 /datum/uplink_item/dangerous/poisonknife
 	name = "Poisoned Knife"
-	desc = "A knife that is made of two razor sharp blades, it has a secret compartment in the handle to store liquids which are injected when stabbing something."
+	desc = "A knife that is made of two razor sharp blades, it has a secret compartment in the handle to store liquids which are injected when stabbing something. Can hold up to fourty units of reagents but comes empty."
```
Comment (Rukofamicom) [inline code/modules/uplink/uplink_items.dm]:
```suggestion
	desc = "A knife that is made of two razor sharp blades, it has a secret compartment in the handle to store liquids which are injected when stabbing something. Can hold up to forty units of reagents but comes empty."
```

**Review by Rukofamicom — COMMENTED** [review]
Aside from typo looks good. 20u is kind of a lot, but it's still only half the capacity of sleepy pen and in a much less conspicuous package.

**Comment by DarnTheMarn** [comment]
The poisoned knife is still pretty bad after this tbh. Why spend any amount of TC on the knife when you can easily grab a syringe gun from medbay, load up a normal syringe, and have the ability to shoot chemicals at range while only losing out on 5u capacity? If the knife is AP, I'm not sure if it is, then you lose out on 10u if you use piercing syringes. Another problem with the knife is that it outs you as 100% being a traitor/excom/incursion/antag unlike a normal syringe gun.

**Comment by Rukofamicom** [comment]
An item being somewhat better does not invalidate the poison knife.

Syringe gun is better at deploying reagents specifically, but the poison knife is a hybrid tool that is both a decent pure damage weapon and also capable of delivering reagents even against protected targets.

**Comment by CydiaLamiales** [comment]
> The item flavor and versatility mentioned above both made the poison knife probably my favorite reagent tool.

It can only transfer five reagents per hit. It costs the same 8 TC as an esword, and comes empty. The pen is so much better than this it's not even funny,

**Comment by Rukofamicom** [comment]
The pen is yes.
Syringe gun no.

**Comment by Fronsis** [comment]
> The pen is yes. Syringe gun no.
> 
> I'd still take knife over sleepy pen just for flavor currently though, it's just a much cooler item.

It also has a different aproach than the sleepy pen/syringe gun, if you start slashing someone they would react differently, play more cautious, stay away, etc giving you more chances to take the upperhand on the scenario and if that fails you can retreat and if they  decide to chase you it's just a matter of time until the person that blindly chased you succumbs/gets slowed in front of you allowing you to have an advantage or they can retreat knowing they've been poisoned, seems like a very good buff and it would be nice to start seeing this more rather than funny ''you feel a tiny prick''

**Review by Rukofamicom — APPROVED** [review]

**Comment by Isy232** [comment]
I'd want the knife down to 5 TC like sleepy pen is/was.

**Comment by CydiaLamiales** [comment]
> I'd want the knife down to 5 TC like sleepy pen is/was.

Why? It costs 1 more because it's reusable and does some damage on its own.

**Comment by Isy232** [comment]
> Why? It costs 1 more because it's reusable and does some damage on its own.

The damage on its own isn't very significant, 15 damage melee weapons are plentiful on the station.
Damaging in the same hit as injecting is good, but not amazing.
If you consider the sleepy pen used to be 5 TC, 45u, refillable and stealthy this poisoned knife is a lot worse.

I agree the old sleepy pen was too powerful, however the current proposed buffed poison knife is still too much of a downgrade from what the sleepy pen used to be.
The poison knife doesn't come with reagents, requiring purchasing the 7 TC poison kit or mixing your own which eats up a significant amount of the limited time you have in a round.

If you'd consider decreasing the cost of the poison kit to 4 TC so combined these cost 10 TC instead of a massive 13 TC that'd also be fine.

**Comment by Rukofamicom** [comment]
> I agree the old sleepy pen was too powerful, however the current proposed buffed poison knife is still too much of a downgrade from what the sleepy pen used to be.

Nothing should be on par with what the sleepy-pen was before the recent nerf.

This is a pretty big buff to the knife, and I have some mild concerns that 20u will prove to be too much to inject in one click.

**Comment by DarnTheMarn** [comment]
What is the attack text for the poisoned knife? Making it the same as a normal knife would make it a bit more stealthy but not too stealthy since if you get busted people can plainly see its a traitor knife.

**Comment by CydiaLamiales** [comment]
> What is the attack text for the poisoned knife? Making it the same as a normal knife would make it a bit more stealthy but not too stealthy since if you get busted people can plainly see its a traitor knife.

The attack messages already are the same as a normal knife.

**Comment by Rukofamicom** [comment]
While the same attack text is used, the poisoned knife does not have a covert name so it cannot pass as any other kind of knife after being seen attacking with it.

I don't think this is an issue for the record, just pointing out that it isn't entirely accurate to say it has the same attack text.

File: code/game/objects/items/kitchen.dm [hunk 2]
```diff
@@ -146,15 +146,15 @@
...
+	var/amount_per_transfer_from_this = 20
 	var/list/possible_transfer_amounts
 	desc = "An infamous knife of syndicate design, it has a tiny hole going through the blade to the handle which stores toxins."
 	materials = null
 
 /obj/item/kitchen/knife/poison/Initialize(mapload)
 	. = ..()
 	create_reagents(40,OPENCONTAINER)
-	possible_transfer_amounts = list(3,5)
+	possible_transfer_amounts = list(5,10,20)
```
Comment (Archanial) [inline code/game/objects/items/kitchen.dm]:
```suggestion
	possible_transfer_amounts = list(5, 10, 20)
```

File: code/game/objects/items/kitchen.dm [hunk 3]
```diff
@@ -146,15 +146,15 @@
 	throwforce = 15
 	throw_speed = 5
 	throw_range = 7
-	var/amount_per_transfer_from_this = 5
+	var/amount_per_transfer_from_this = 20
```
Comment (Archanial) [inline code/game/objects/items/kitchen.dm]: 20 is too much in my opinion. I'd reduce TC cost further and keep it at 10.
20 means you would transfer the entire amount in just 2 hits, which is too quick.

**Review by Archanial — CHANGES_REQUESTED** [review]

**Review by Archanial — APPROVED** [review]
cool :sunglasses:

**Pull request merged**